hipify-clang/src/Cuda2Hip.cpp

```diff
@@ -620,289 +620,50 @@ class Cuda2HipCallback : public MatchFinder::MatchCallback, public Cuda2Hip {
           }
-        }
-        else {
-          std::string msg = "the following reference is not handled: '" + name + "' [typedef var ptr].";
-          printHipifyMessage(*SM, sl, msg);
-        }
       }
-      return true;
-    }
-    return false;
-  }
 
-  bool cudaStructVar(const MatchFinder::MatchResult &Result) {
-    if (const VarDecl *structVar = Result.Nodes.getNodeAs<VarDecl>("cudaStructVar")) {
-      QualType QT = structVar->getType();
-      // ToDo: find case-studies with types other than Struct.
-      if (QT->isStructureType()) {
-        std::string name = QT.getTypePtr()->getAsStructureType()->getDecl()->getNameAsString();
-        TypeLoc TL = structVar->getTypeSourceInfo()->getTypeLoc();
-        SourceLocation sl = TL.getUnqualifiedLoc().getLocStart();
-        SourceManager *SM = Result.SourceManager;
-        const auto found = CUDA_TO_HIP_RENAMES.find(name);
-        if (found != CUDA_TO_HIP_RENAMES.end()) {
-          updateCounters(found->second, name);
-          if (!found->second.unsupported) {
-            StringRef repName = found->second.hipName;
-            Replacement Rep(*SM, sl, name.size(), repName);
-            FullSourceLoc fullSL(sl, *SM);
-            insertReplacement(Rep, fullSL);
-          }
-        }
-        else {
-          std::string msg = "the following reference is not handled: '" + name + "' [struct var].";
-          printHipifyMessage(*SM, sl, msg);
-        }
+      // Do we have a replacement for this type?
+      const auto found = CUDA_TO_HIP_RENAMES.find(typeName);
+      if (found == CUDA_TO_HIP_RENAMES.end()) {
+          return false;
       }
-      return true;
-    }
-    return false;
-  }
 
-  bool cudaStructVarPtr(const MatchFinder::MatchResult &Result) {
-    if (const VarDecl *structVarPtr = Result.Nodes.getNodeAs<VarDecl>("cudaStructVarPtr")) {
-      const Type *t = structVarPtr->getType().getTypePtrOrNull();
-      if (t) {
-        TypeLoc TL = structVarPtr->getTypeSourceInfo()->getTypeLoc();
-        SourceLocation sl = TL.getUnqualifiedLoc().getLocStart();
-        SourceManager *SM = Result.SourceManager;
-        StringRef name = t->getPointeeCXXRecordDecl()->getName();
-        const auto found = CUDA_TO_HIP_RENAMES.find(name);
-        if (found != CUDA_TO_HIP_RENAMES.end()) {
-          updateCounters(found->second, name.str());
-          if (!found->second.unsupported) {
-            StringRef repName = found->second.hipName;
-            Replacement Rep(*SM, sl, name.size(), repName);
-            FullSourceLoc fullSL(sl, *SM);
-            insertReplacement(Rep, fullSL);
-          }
-        } else {
-          std::string msg = "the following reference is not handled: '" + name.str() + "' [struct var ptr].";
-          printHipifyMessage(*SM, sl, msg);
-        }
+      const hipCounter& hipCtr = found->second;
+      if (hipCtr.unsupported) {
+          llvm::errs() << "[HIPIFY] Unsupported type: " << typeName << "\n";
```

Comment:
Better to give FullSourceLoc for the found unsupported type.

Reply:
Oh, right. I was meaning to port all of these over to the Diagnostics API anyway.

Reply:
What about testing? What sources were tested?

Reply:
In addition to my CUDA code I'm trying to get working, I wrote the following contrived program to try to expose failures in my changed code:
https://gist.github.com/ChrisKitching/827dbc17acd939c3cd4f9b898006cc02

We should probably add some more testcases to the `lit` testsuite - current coverage there is relatively poor. While the automated test suite fails (and, indeed, it seems broken in general), that input file yields correct-looking output, too.

Reply:
(Fixed both your review comments now, too)

Reply:
We will definitely extend testing. I thought to add also golden files comparison. And btw lit testing is not the demand. As for you tremendous changes in this pull request, I'd like you to perform more testing manually and diff against current hipify-clang, or we may do it together.
1.	https://github.com/ROCm-Developer-Tools/HIP/blob/master/samples/0_Intro/square/square.cu
2.	https://github.com/ROCm-Developer-Tools/HIP/files/1389647/Interac-v2.zip
3.	https://github.com/ROCm-Developer-Tools/HIP/blob/master/tests/src/cudaRegister.cu
4.	https://github.com/AMDComputeLibraries/hip/blob/master/examples/rodinia_3.0/cuda/backprop/backprop_cuda.cu
5.	CUDA SDK samples, I’ll select a few.

Reply:
> https://github.com/ROCm-Developer-Tools/HIP/files/1389647/Interac-v2.zip

This does not contain any CUDA.

Reply:
> https://github.com/AMDComputeLibraries/hip/blob/master/examples/rodinia_3.0/cuda/backprop/backprop_cuda.cu

That's a 404. Is that an internal AMD repo I don't get to see, perhaps?